modules/frontend/pipeline/async_weight_middleware.go

```diff
@@ -114,12 +114,17 @@ func (c weightRequestWare) setTraceQLWeight(req Request) {
 		if c.Op != traceql.OpNone {
 			conditions++
 		}
+
+		if c.Attribute.Intrinsic == traceql.IntrinsicNone && c.Attribute.Scope == traceql.AttributeScopeNone {
```

Comment:
what are we trying to catch here?

Reply:
oh, you are looking for unscoped attributes. it may be safer to test `spanRequest.AllConditions` at the bottom?

i'm fine with this, but perhaps do a bit of testing before we merge?

Reply:
Do you mean something like?:

```go
if heavyQuery  && !spanRequest.AllConditions{
		req.SetWeight(c.weights.TraceQLSearchWeight + 1)
	}
```

Reply:
apologies, i meant to edit this comment and got sidetracked. `AllConditions` will only be true if its a single spanset selector will all conditions `&&`'ed. It will be true even with unscoped attributes, so it will be true on both of these:

```
{ span.foo = "bar" && resource.baz = "bat" }
{  .foo = "bar" && .baz = "bat" }
```

If AllConditions is false -OR- there are unscoped attributes the querier is a heavier query b/c it cannot be evaluated in the fetch layer and requires the engine.

It's possible that both unscoped attributes and AllConditions false should result in `TraceQLSerachWeight + 1` but would need some testing.